third_party/nvfuser/csrc/ir_utils.h

```diff
@@ -364,5 +364,8 @@ TORCH_CUDA_CU_API bool isTorchGatherLookupTv(const Val* tv);
 
 TORCH_CUDA_CU_API std::string varName(const Val* val);
 
+// Check if a tensor is resized as part of  its root to rfactor transformations
+bool hasResizedRfactor(const TensorView* tv);
```

Comment:
Resize can exist after rfactor. Suppose:

```
tv1 = pad(tv0, {...});
```

And when indexing this expression, `tv0` is transformed as `tv1`, which includes replaying the resize expr of `tv1`. So it can exist outside of the rfactor transformation, but it needs to be done carefully. Resizing a producer tensor is fine as the transformation is temporary and it doesn't affect how it is actually calculated. However, we shouldn't use `resize` as a scheduling primitive like `merge` and `split`, and there's no validation for that.

I think what we could do is to do some validation at the beginning of lowering. At that point, resize should only be used between root and rfactor.

Reply:
Related comment: https://github.com/csarofeen/pytorch/pull/2480#discussion_r1122746605

Reply:
Validation added: 3ed4797d13e34048ef5ffae15786cb8d9164dd6d

Reply:
Makes sense, but scares me in the way that it also feels a lot like broadcast resolution.